src/safeds/data/image/containers/_image.py

```diff
@@ -305,6 +305,49 @@ def adjust_brightness(self, factor: float) -> Image:
         image_copy._image = ImageEnhance.Brightness(image_copy._image).enhance(factor)
         return image_copy
 
+    def add_gaussian_noise(self, standard_deviation: float, opacity: float = 1.0) -> Image:
+        if standard_deviation < 0:
+            raise ValueError("Standard deviation has to be 0 or bigger.")
+
+        if opacity < 0 or opacity > 1:
+            raise ValueError("Opacity has to be between 0 and 1.")
+
+        if opacity == 0:
+            warnings.warn(
+                "Opacity is 0, this will not make changes to the image.",
+                UserWarning,
+                stacklevel=2,
+            )
+
+        image_copy = copy.deepcopy(self)
+        img_noise = PIL.Image.effect_noise((image_copy.width, image_copy.height), standard_deviation)
+        image_copy._image = PIL.Image.blend(image_copy._image.convert("L"), img_noise.convert("L"), 0.5)
```

Comment:
`opacity` is never used here.

Reply:
I would also not expect the image to be converted to greyscale.

Reply:
this is fixed now. had to do some kind of coversion for the blend method to work, now i went with an rgb conversion